scipy/linalg/tests/test_lapack.py

```diff
@@ -559,3 +559,50 @@ def test_sgesdd_lwork_bug_workaround():
     assert_equal(returncode, 0,
                  "Code apparently failed: " + p.stdout.read())
 
+
+class TestSytrd(object):
+    def test_sytrd(self):
+        for dtype in REAL_DTYPES:
+            # "random" symmetric matrix
+            A = np.array([
```

Comment:
If you create `A` via 

    n = 3
    A = np.zeros((n, n), dtype=dtype)
    A[np.triu_indices_from(A)] = np.arange(1, 2*n+1, dtype=dtype)

(or typing out manually if you wish), you can quickly test for `lower=1` behavior for `n` of your choice since it will give the matrix and it's diagonal back together with zero arrays for a diagonal matrix. Then it can continue with `lower=0` case for the precise calculation.

The comment can be worded as `# some upper triangular array` to avoid confusion about the randomness too.